app/client/components/register/AccomodationFieldGroup.tsx

```diff
@@ -0,0 +1,122 @@
+import { FieldHandler } from "@reactivated";
+import React, { useState } from "react";
+import { WidgetHandler } from "reactivated/dist/forms";
+import { DjangoFormsWidgetsCheckboxInput } from "reactivated/dist/generated";
+import { BasicFormField } from "../forms/BasicFormField";
+
+interface AccomodationFieldGroupProps {
+  dinnerField: WidgetHandler<DjangoFormsWidgetsCheckboxInput>;
+  accomodationField: WidgetHandler<DjangoFormsWidgetsCheckboxInput>;
+  breakfastField: WidgetHandler<DjangoFormsWidgetsCheckboxInput>;
+
+  onCostChange: (
+    breakfast: boolean,
+    dinner: boolean,
+    accomodation: boolean,
+  ) => void;
+}
+
+interface GroupState {
+  accomodation: {
+    checked: boolean;
+  };
+  breakfast: {
+    disabled: boolean;
+    checked: boolean;
+  };
+  dinner: {
+    disabled: boolean;
+    checked: boolean;
+  };
+}
+
+export const AccomodationFieldGroup = ({
+  dinnerField,
+  accomodationField,
+  breakfastField,
+  onCostChange,
+}: AccomodationFieldGroupProps) => {
+  const [groupState, setGroupState] = useState<GroupState>({
```

Comment:
It seems like a perfect use case for reducer

Reply:
Done